AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -0,0 +1,612 @@
+      :: k := activeItem.EncryptionContext[k];
+
+    // To Preserve Unexpected/un-modeled Attributes.
+    // We need to copy them from inferredOriginalEC to terminalEC.
+    // Which means we need to select those members without a prefix,
+    // and copy them over to terminal.
+    var unexpectedEC := PrefixUtils.FilterMapForKeysThatDoNotBeginWithPrefix(
+      prefix := Structure.ENCRYPTION_CONTEXT_PREFIX,
+      aMap := inferredOriginalEC
+    );
+    assert unexpectedEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES;
+
+    var terminalEC?: Option<KeyStoreTypes.EncryptionContextString> := None;
+    if (input.Mutations.TerminalEncryptionContext.Some?) {
+
+      var terminalEC := PrefixUtils.AddingPrefixToKeysOfMapDoesNotCreateCollisions(
+        prefix := Structure.ENCRYPTION_CONTEXT_PREFIX,
+        aMap := input.Mutations.TerminalEncryptionContext.value
+      ) + unexpectedEC;
+      // ValidateInitializeMutationInput SHOULD take care of this Need, but Dafny is struggling
+      // TODO-Mutations-FF : Replace runtime check with Lemma.
+      // See https://github.com/aws/aws-cryptographic-material-providers-library/pull/750#discussion_r1777654751
+      :- Need(
+        terminalEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES,
+        Types.KeyStoreAdminException(message:="Terminal Encryption Context contains a reserved word!")
+      );
+      terminalEC? := Some(terminalEC);
+      assert terminalEC.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES;
+    }
+
+    assert KmsArn.ValidKmsArn?(activeItem.KmsArn);
+    var MutationToApply := StateStrucs.MutationToApply(
+      Identifier := input.Identifier,
+      Original := StateStrucs.MutableProperties(
+        kmsArn := activeItem.KmsArn,
+        customEncryptionContext := inferredOriginalEC
+      ),
+      Terminal := StateStrucs.MutableProperties(
+        kmsArn := input.Mutations.TerminalKmsArn.UnwrapOr(activeItem.KmsArn),
+        customEncryptionContext := terminalEC?.UnwrapOr(inferredOriginalEC)
+      ),
+      ExclusiveStartKey := None,
+      UUID := mutationCommitmentUUID,
+      CreateTime := timestamp,
+      Input := input.Mutations,
+      CommitmentCiphertext := [0], // TODO-Mutations-GA Create Commitment Ciphertext
+      IndexCiphertext := [0] // TODO-Mutations-GA Create Index Ciphertext
+    );
+
+    assert MutationToApply.Original.customEncryptionContext.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES;
+    assert MutationToApply.Terminal.customEncryptionContext.Keys !! Structure.BRANCH_KEY_RESTRICTED_FIELD_NAMES;
+    assert MutationToApply.ValidState();
+    // -= BEGIN Version Active Branch Key
+    // --= Validate Active Branch Key
+    var verifyActive? := Mutations.VerifyEncryptedHierarchicalKey(
+      item := activeItem,
+      keyManagerStrategy := input.keyManagerStrategy,
+      localOperation := "InitializeMutation"
+    );
+    if (verifyActive?.Fail?) {
+      return Failure(verifyActive?.error);
+    }
```

Comment:
```suggestion
    var verifyActive :- Mutations.VerifyEncryptedHierarchicalKey(
      item := activeItem,
      keyManagerStrategy := input.keyManagerStrategy,
      localOperation := "InitializeMutation"
    );
```

Reply:
No, I have to revert this back to the way it was.
Because `Verify` returns an Outcome,
but the calling method returns a Result.